Tutorial/brusselator/analysis.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Analysis code for the Brusselator application.
+ * Reads variable u_real, u_imag, v_real, v_imag, and computes the norm of U and V.
+ * Writes the variables and their computed norm to an ADIOS2 file.
+ *
+ * Kshitij Mehta
+ *
+ * @TODO:
+ *      - Error checks. What is vector resizing returns an out-of-memory error? Must handle it
+ *      - Turn ADIOS2 Debug to ON
+ */
+#include <iostream>
+#include <stdexcept>
+#include <cstdint>
+#include <cmath>
+#include "adios2.h"
+
+/*
+ * Function to compute the norm of an array
+ */
+std::vector<double> compute_norm (std::vector<double> real_part, std::vector<double> imag_part, int dims) {
```

Comment:
I'd make this function a template so it can be used for either floats or doubles. No need to carry the dims calling .size() on any vector will give you its size. Also, add const and pass by reference (otherwise this function copies both vectors):
Something like:
```
template <class T> 
std::vector<T> compute_norm(const std::vector<T>& real, const std::vector<T>& imag){

    if(real.size() != imag.size()) 
    {     // needs #include <stdexcept>
          throw std::invalid_argument("ERROR: real and imag parts have different sizes\n"); 
    }
    std::vector<T> norm( real.size() );  // allocate at creation
    for (auto i = 0; i < real.size(); ++i )
    {
        //needs #include <cmath>, put 2. , not just 2
        norm[i] = std::sqrt( std::pow(real[i], 2.) + std::pow(imag[i], 2.) );
    }
    return norm;
}
```